034-deprecate-and-remove-mirror-maker-2-extensions.md

```diff
@@ -0,0 +1,46 @@
+# Deprecate and remove Mirror Maker 2 extensions
+
+## Current situation
+
+When Mirror Maker 2 was first introduced, it was missing a policy for replicating topics between two Kafka clusters without changing their names.
+We created the [_Mirror Maker 2 Extensions_ project](https://github.com/strimzi/mirror-maker-2-extensions) which contains the `IdentityReplicationPolicy` policy.
+This policy makes sure that topics are not renamed while being mirrored.
+The `IdentityReplicationPolicy` is the only _extension_ we provide ass part of this project.
+
+In Kafka 3.0.0, Mirror Maker 2 introduces its own `IdentityReplicationPolicy`.
+This was done in [KAFKA-9726](https://issues.apache.org/jira/browse/KAFKA-9726).
+This policy provides the same features as Strimzi's own `IdentityReplicationPolicy`.
+It does not seem to make sense to maintain our own policy in parallel to Apache Kafka.
+
+## Proposal
+
+After dropping support for Kafka versions older than 3.0.0 (expected to happen in Strimzi 0.28.0), we should deprecate our own policy and migrate users to the Kafka's identity replication policy.
+
+Following steps should be taken as part of the 0.28.0 release:
+* Update our documentation to use Kafka's `IdentityReplicationPolicy`.
+* Update our examples to use Kafka's `IdentityReplicationPolicy`.
+* Update the `CHANGELOG.md` file to indicate the deprecation.
+* Release new version of the MM2 Extensions which would instead of implementing our own logic just extend Kafka's `IdentityReplicationPolicy`.
+  That should ensure exactly the same behavior from both policies.
+  It would also make sure that any users still using `io.strimzi.kafka.connect.mirror.IdentityReplicationPolicy` do not have any problems
+* After the release, clearly deprecate and archive the `mirror-maker-2-extensions` repository.
+  (If we ever need any other MM2 extension in the future, we can always unarchive it)
+
+Even in 0.28 and newer releases, will still include the `mirror-maker-2-extensions` JARs to ensure compatibility.
+The JAR will be removed only in Strimzi 0.32.0 or Strimzi 1.0.0 (whichever comes first).
```

Comment:
What's the rationale for 0.32.0?

Reply:
The _rational_ is 4 releases. But it is basically a random pick without any real justification why exactly 4. If you feel it should be something else I don't think there is a major issue to change it. I just had to make the cut somewhere.

Reply:
No objection to 4 specifically, just to majik numbers :-)